aea/protocols/base.py

```diff
@@ -83,6 +84,26 @@ def __init__(self, _body: Optional[Dict] = None, **kwargs):
         except Exception as e:  # pylint: disable=broad-except
             _default_logger.error(e)
 
+    def json(self) -> dict:
+        """Get json friendly str representation of the message."""
+        return {
+            "to": self._to,
+            "sender": self._sender,
+            "body": b64encode(self.encode()).decode("utf-8"),
+        }
+
+    @classmethod
+    def from_json(cls, data: dict) -> "Message":
+        """Construct message instance from json data."""
+        instance = cls.decode(b64decode(data["body"]))
+        sender = data["sender"]
+        if sender:
+            instance.sender = sender
+        to = data["to"]
+        if to:
+            instance.to = to
```

Comment:
let's use .get instead

Reply:
Or at least catch key error and print a more useful exception, like data corrupted in db or something like this

Reply:
done